### PR TITLE
Small Performance Tweaks Round 1

### PR DESCRIPTION
**Pull request by Railboy**
Overview
---
First round of small performance tweaks. These were all problems we spotted in the profiler while testing. Tried to avoid breaking changes or heavy refactoring.

Changes
---
- Set new default for line steps in pointer prefabs and updated pointer inspector to give a perf warning.
- Disabled auto-sync transforms in physics settings. This helped with physics spikes while using manipulation scripts. [See docs.](https://docs.unity3d.com/ScriptReference/Physics-autoSyncTransforms.html)
- Added a new performance-oriented setting in line data providers.
- Replaced a lot of Vector3 math in very hot code, especially RayStep.
- Removed some unnecessary allocations.

### CODE REVIEW

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/LinePointer.cs [hunk 1]
```diff
@@ -15,6 +15,11 @@ namespace Microsoft.MixedReality.Toolkit.SDK.UX.Pointers
     [RequireComponent(typeof(DistorterGravity))]
     public class LinePointer : BaseControllerPointer
     {
+        [Range(2, 50)]
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/LinePointer.cs]: Should we go ahead and start this at `1`, or is a minimum of 2 required? (thinking of cases like a straight line)

File: Assets/MixedRealityToolkit.SDK/Inspectors/UX/Pointers/LinePointerInspector.cs [hunk 2]
```diff
@@ -18,6 +18,7 @@ public class LinePointerInspector : BaseControllerPointerInspector
         private SerializedProperty lineRenderers;
 
         private bool linePointerFoldout = true;
+        private const int maxRecommendedLinecastResolution = 25;
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.SDK/Inspectors/UX/Pointers/LinePointerInspector.cs]: The [Tooltip on LinePointer](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/3479/files#diff-42582bc1e2308bec1f932a22d56c0488R20) recommends 20 as the max for perf reasons. Should we standardize on 20 or 25 between here and there?

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/LinePointer.cs (hunk 1 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/LinePointer.cs]: also, is 50 still too many?

File: Assets/MixedRealityToolkit.SDK/Inspectors/UX/Pointers/LinePointerInspector.cs [hunk 3]
```diff
@@ -42,12 +43,19 @@ public override void OnInspectorGUI()
             if (linePointerFoldout)
             {
                 EditorGUI.indentLevel++;
+
+                int lineCastResolutionValue = lineCastResolution.intValue;
+                if (lineCastResolutionValue > maxRecommendedLinecastResolution)
+                {
+                    EditorGUILayout.LabelField("Note: values above " + maxRecommendedLinecastResolution + " should only be used when your line is expected to be highly non-uniform.", EditorStyles.miniLabel);
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit.SDK/Inspectors/UX/Pointers/LinePointerInspector.cs]: can you define "highly non-uniform"?

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/LinePointer.cs (hunk 1 above)
Comment (Railboy) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/LinePointer.cs]: In 95% of cases probably yes. There are cases where 50 could be useful - eg a really weird or inventive spline pointer. The inspector does give you a warning if you set it above 25 (soon to be 20).

File: Assets/MixedRealityToolkit.SDK/Inspectors/UX/Pointers/LinePointerInspector.cs (hunk 3 above)
Comment (Railboy) [reply Assets/MixedRealityToolkit.SDK/Inspectors/UX/Pointers/LinePointerInspector.cs]: A controller styled after a rope, whip or tentacle. Something with lots of bends that needs to 'thread the needle' when detecting collisions.

**Comment by davidkline-ms** [comment]
@Railboy, what was the reason for reverting the cached collider change?

**Comment by Railboy** [comment]
@davidkline-ms Caching the collider bounds was causing some unexpected behavior.

**Pull request merged**